File: backend/app/models/notification.py
from sqlalchemy import Column, Integer, String, Boolean, ForeignKey, DateTime
from sqlalchemy.sql import func
from app.db.database import Base

class Notification(Base):
    __tablename__ = "notifications"

    id = Column(Integer, primary_key=True, index=True)
    recipient_id = Column(Integer, ForeignKey("users.id"), nullable=False) # Người nhận thông báo
    sender_id = Column(Integer, ForeignKey("users.id"), nullable=False)    # Người gây ra hành động
    
    # Loại thông báo: 'like', 'comment', 'friend_request'
    type = Column(String(50), nullable=False)
    
    # ID của đối tượng liên quan (ví dụ id của bài viết bị like)
    target_id = Column(Integer, nullable=True) 
    
    is_read = Column(Boolean, default=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

File: backend-java/src/main/java/com/example/backend_java/entity/Notification.java
package com.example.backend_java.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Người nhận thông báo
    @Column(name = "recipient_id", nullable = false)
    private Long recipientId;

    // Người gây ra hành động
    @Column(name = "sender_id", nullable = false)
    private Long senderId;

    // Loại thông báo: 'like', 'comment', 'friend_request'
    @Column(length = 50, nullable = false)
    private String type;

    // ID của đối tượng liên quan (ví dụ: id bài viết bị like)
    @Column(name = "target_id")
    private Long targetId;

    @Column(name = "is_read")
    @Builder.Default
    private Boolean isRead = false;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;
}
